### PR TITLE
add typescript and vite

### PR DESCRIPTION
**Pull request by apucontilde**
Screenshot 2023-07-24 160131

### CODE REVIEW

**Comment by jvsqzj** [comment]
vite?

File: src/utils.ts [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
-      timeout = null
-      if (!immediate) func.apply(context, args)
-    }
-    const callNow = immediate && !timeout
-    clearTimeout(timeout)
-    timeout = setTimeout(later, wait)
-    if (callNow) func.apply(context, args)
+export function debounce<F extends (...args: Parameters<F>) => ReturnType<F>>(
+  func: F,
+  wait: number,
```
Comment (gabygarro) [inline src/utils.ts]: trailing space

**Review by gabygarro — CHANGES_REQUESTED** [review]
pls make the linter work

File: src/index.tsx [hunk 2]
```diff
@@ -5,16 +5,16 @@ import "@fontsource/roboto/400.css"
 import "@fontsource/roboto/500.css"
 import "@fontsource/roboto/700.css"
 import "./index.css"
-import App from "./App"
+import App from "./App.tsx"
```
Comment (gabygarro) [inline src/index.tsx]:
```suggestion
import App from "./App"
```

Comment (apucontilde) [reply src/index.tsx]: done!

File: src/utils.ts (hunk 1 above)
Comment (apucontilde) [reply src/utils.ts]: done

**Review by gabygarro — APPROVED** [review]

**Pull request merged**